### PR TITLE
Improve contact details page

### LINKED ISSUE #6: Frontend team should add something like getIcon to MultaccItem
https://github.com/multacc/multacc/blob/86ea010ff47f5d749bf92c2cf569ee4f7d1f9a8b/lib/contact_types/item.dart#L25-L30

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `todo` comment in 86ea010ff47f5d749bf92c2cf569ee4f7d1f9a8b when #4 was merged. cc @multacc.

### PR DESCRIPTION
**Pull request by mayank99**
- Replaced draggable sheet with new package to fix snapping
- Added search support for contacts
- Added avatars with fallback icon
- Implemented launching phone and email
- Added settings page (with a selection for preferred phone app)
- Added toSet() to prevent duplicating phone numbers
- Lots of refactoring and cleanup 🍸 

Closes #6, #33, #35, #50, #53

### CODE REVIEW

File: lib/items/phone.dart [hunk 1]
```diff
@@ -1,25 +1,57 @@
...
+      case 'default':
+      default:
+        launch('tel:$phone');
+        break;
+    }
   }
 
   bool isLaunchable() => true;
+
+  @override
```
Comment (TortoiseWrath) [inline lib/items/phone.dart]: @override annotations for methods with no base implementation are gross, change my mind

```suggestion
```

File: lib/pages/contacts/contact_model.dart [hunk 2]
```diff
@@ -34,7 +34,7 @@ class MultaccContact extends Contact {
     clientKey = identifier; // Key in client-side database
     serverKey = null; // Key in server-side database
     multaccItems = [
-      ...phones.map((item) => PhoneItem.fromItem(item)), // create PhoneItems from phone Items
+      ...phones.toSet().map((item) => PhoneItem.fromItem(item)), // create PhoneItems from phone Items
```
Comment (TortoiseWrath) [inline lib/pages/contacts/contact_model.dart]: beautiful

File: lib/items/phone.dart (hunk 1 above)
Comment (mayank99) [reply lib/items/phone.dart]: no u

Comment (TortoiseWrath) [reply lib/items/phone.dart]: he says after adding it to one of his three implementations of getIcon

Comment (mayank99) [reply lib/items/phone.dart]: To be fair, two of those were written manually. This one was generated by the Flutter extension. All I did was type `getIc` and hit enter.

**Pull request merged**